### ISSUE TITLE
fmt: respect one-liner `if` expressions even when an `else` is present

### ISSUE THREAD

**Issue by anderseknert** [open-policy-agent/opa#6733]
Whether `else` should be moved to a line of its own at all is debatable, but I think the formatter at minimum should respect simple one-liner constructs when `else` is involved:

This:
```rego
foo if input.bar else := false
```
Currently formats to this:
```rego
foo if input.bar

else := false
```
Which just makes the `else` look entirely detached from the `if`.

**Comment by sspaink** [comment]
I agree, the newline doesn't look good. There is an existing test file though that disagrees: https://github.com/open-policy-agent/opa/blob/main/v1/format/testfiles/v1/test_if_else.rego.formatted

The examples in `test_if_else.rego.formatted` aren't all one-liners but they don't look great to me with the extra space either, what do you think?
